File: frontend/src/schema/room.ts
export interface RoomBase {
    key: string;
    name: string;
    message: string;
}

export interface CreateRoomRequest extends Omit<RoomBase, "key" | "name"> {
    user_id: number;
    key?: string | null;
    name?: string | null;
}

export interface UpdateRoomRequest extends Omit<RoomBase, "key" | "name" | "message"> {
    key?: string | null;
    name?: string | null;
    message?: string | null;
}

export interface RoomResponse extends RoomBase {
    room_id: number;
    click: number;
}

export interface DeleteRoomResponse {
    room_id: number;
    is_deleted: boolean;
}

export interface CheckRoomResponse {
    is_available: boolean;
}

File: backend/src/schema/room.py
from pydantic import BaseModel
from typing import Optional

class RoomBase(BaseModel):
    key: str
    name: str
    message: str

class CreateRoomRequest(RoomBase):
    user_id: int
    key: Optional[str] = None
    name: Optional[str] = None

class UpdateRoomRequest(RoomBase):
    key: Optional[str] = None
    name: Optional[str] = None
    message: Optional[str] = None

class RoomResponse(RoomBase):
    room_id: int
    click: int

class DeleteRoomResponse(BaseModel):
    room_id: int
    is_deleted: bool

class CheckRoomResponse(BaseModel):
    is_available: bool